crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -629,8 +856,33 @@ impl<'a> SimpleTokenizer<'a> {
                             self.cursor = savepoint;
                             SimpleTokenKind::Other
                         }
+                    } else if is_any_operator_token(c) {
+                        // This could be a single-token token, like `+` in `x + y`, or a
+                        // multi-character token, like `+=` in `x += y`. It could also be a sequence
+                        // of multi-character tokens, like `x ==== y`, which is invalid, _but_ it's
+                        // important that we produce the same token stream when lexing backwards as
+                        // we do when lexing forwards. So, identify the range of the sequence, lex
+                        // forwards, and return the last token.
+                        let mut cursor = self.cursor.clone();
+                        cursor.eat_back_while(is_any_operator_token);
+
+                        let token_len = cursor.token_len();
+                        let range = TextRange::at(self.back_offset - token_len, token_len);
+
+                        let cursor = Self::new(self.source, range);
+                        if let Some(token) = cursor.last() {
```

Comment:
Naming this `cursor` is confusing because it is the lexer. Rename to `forward_lexer`?